packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -118,7 +126,95 @@ export function getMiniflareOptions(
 	viteConfig: vite.ResolvedConfig,
 	viteDevServer: vite.ViteDevServer,
 ): MiniflareOptions {
-	const workers = Object.values(normalizedPluginConfig.workers).map(
+	const assetWorkers: Array<WorkerOptions> = [
```

Comment:
Should this stuff not be configurable?
I.e. not adding this stuff for Workers that do not have assets configured?

If I am building a Worker that does not have assets then I would not expect `vite dev` to serve assets from my local file system ahead of my Worker code.

Reply:
> Should this stuff not be configurable? I.e. not adding this stuff for Workers that do not have assets configured?

My instinct is that an assets only project (static sites etc.) should be zero config by default. If we needed to make it configurable, I would therefore argue that it should opt-out rather than opt-in. I'm not yet convinced that it's actually necessary though and think that we can consider that with broader questions about what the API should look like.

> If I am building a Worker that does not have assets then I would not expect `vite dev` to serve assets from my local file system ahead of my Worker code.

I think that's where we differ because that would always be my expectation about how Vite would work. This already happens before this PR and will continue to happen unless we remove the default middlewares. We may want to go with that approach in the future in order to allow the Worker to sit in front of all assets and for the plugin to take control all asset handling. We would need to look at the implications carefully though, particularly in relation to full-stack frameworks.

Reply:
So will `vite build` always output a Worker with Assets turned on, to account for the potential assets that are served by default in `vite dev`?

Reply:
For standard Vite, the things that are included in the client build are anything that's imported by the entry file (`index.html` by default) and/or its sub-dependencies (including static assets) and the contents of the `public` directory (the name of this directory is configurable). This PR doesn't implement the build side of this but my thinking was that the build would check if the client entry file exists and, if it does, build the client environment and reference its output as the assets directory.

We need to have a chat about using the assets binding to load non-HTML assets and ensuring that these are included in the build. That will be a subsequent PR.